### ISSUE TITLE
[Docs] Embedded node client should disable HTTP

### ISSUE THREAD

**Issue by pickypg** [elastic/elasticsearch#11952]
The Java API documentation talks about the creation of an embedded node client, but it does not talk about the negatives. One oft-surprising negative is that, because it's a full node, it also provides an HTTP port for access.

For embedded client nodes, this should be disabled by default. For standalone client nodes (the main use case for client nodes), this should be on by default.

```java
Node node =
    nodeBuilder()
        .settings(ImmutableSettings.settingsBuilder().put("http.enabled", false))
        .client(true)
    .node();

Client client = node.client();
```

**Comment by dadoonet** [comment]
@pickypg you wrote almost everything ! Why not sending a PR ?

Wanna do it ?

**Comment by pickypg** [comment]
Closed by 8efa18e6167e276dd220b60c34b516d05252bc22 (master), 2461e913b717cf2c06f58a615ae8aa65a7a70a7f (1.7), d4b7e12d1cf57899db990690413d49a26fd37370 (1.6)
